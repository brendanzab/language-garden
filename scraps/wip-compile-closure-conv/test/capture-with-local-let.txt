let a : Int := 1;
let f : Int -> Int :=
  fun (x : Int) =>
    let y : Int := x + a;
    y + 3;

f 100
