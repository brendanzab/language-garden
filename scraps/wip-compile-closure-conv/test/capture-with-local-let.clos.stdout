let a : Int := 1;
let f : Int -> Int :=
  clos(
    fun (env : (Int,)) (x : Int) => let y : Int := #add x env.0; #add y 3,
    (a,)
  );
f 100
